lib/improver/tests/blending/weights/test_ChooseWeightsLinearFromDict.py

```diff
@@ -0,0 +1,513 @@
+
+
+class Test__repr__(IrisTest):
+    """Test the __repr__ method."""
+
+    def test_basic(self):
+        """Test the repr function formats the arguments correctly"""
+        weighting_coord_name = "forecast_period"
+        config_coord_name = "model_configuration"
+        config_dict = {"uk_det": {"forecast_period": [7, 12, 48, 54],
+                                  "weights": [0, 1, 1, 0],
+                                  "units": "hours"}}
+        plugin = ChooseWeightsLinearFromDict(
+            weighting_coord_name, config_coord_name, config_dict)
+        result = str(plugin)
+        expected = ("<ChooseWeightsLinearFromDict "
+                    "config_dict = {}, "
+                    "weighting_coord_name = forecast_period, "
+                    "config_coord_name = model_configuration, "
+                    "weights_coord_name = weights>".format(str(config_dict)))
+        self.assertEqual(result, expected)
+
+    def test_alternative_name(self):
+        """Test the repr function formats the arguments correctly"""
+        weighting_coord_name = "forecast_period"
+        config_coord_name = "model_configuration"
+        config_dict = {"uk_det": {"forecast_period": [7, 12, 48, 54],
+                                  "alternative_name": [0, 1, 1, 0],
+                                  "units": "hours"}}
+        weights_coord_name = "alternative_name"
+        plugin = ChooseWeightsLinearFromDict(
+            weighting_coord_name, config_coord_name, config_dict,
+            weights_coord_name=weights_coord_name)
+        result = str(plugin)
+        expected = ("<ChooseWeightsLinearFromDict "
+                    "config_dict = {}, "
+                    "weighting_coord_name = forecast_period, "
+                    "config_coord_name = model_configuration, "
+                    "weights_coord_name = alternative_name>".format(
+                        str(config_dict)))
+        self.assertEqual(result, expected)
+
+
+class Test__check_config_dict(IrisTest):
+    """Test the _check_config_dict method. """
+
+    def test_dictionary_key_mismatch(self):
+        """Test whether there is a mismatch in the dictionary keys. As
+        _check_config_dict is called within the initialisation,
+        _check_config_dict is not called directly."""
+        config_dict = {"uk_det": {"forecast_period": [7, 12, 48, 54],
+                                  "weights": [0, 1, 0],
+                                  "units": "hours"}}
+        weighting_coord_name = "forecast_period"
+        config_coord_name = "model_configuration"
+
+        msg = ('These items in the configuration dictionary')
+        with self.assertRaisesRegex(ValueError, msg):
+            ChooseWeightsLinearFromDict(
+                weighting_coord_name, config_coord_name,
+                config_dict=config_dict)
+
+    def test_dictionary_key_match(self):
+        """Test whether that the dictionary keys match in length as
```

Comment:
Grammar. How about "Test no exception is raised when..."

Reply:
I've corrected this docstring.